File: 品味Java的21種設計模式/dp_java/mediator/example1/ConcreteColleagueA.java
package cn.javass.dp.mediator.example1;
/**
 * 具体的同事类A
 */
public class ConcreteColleagueA extends Colleague {
	public ConcreteColleagueA(Mediator mediator) {
		super(mediator);
	}
	/**
	 * 示意方法，执行某些业务功能
	 */
	public void someOperation() {
		//在需要跟其他同事通信的时候，通知中介者对象
		getMediator().changed(this);
	}
}

File: 品味Java的21種設計模式/dp_cpp/mediator/example1/ConcreteColleagueA.cpp
#include "ConcreteColleagueA.h"

namespace cn {
namespace javass {
namespace dp {
namespace mediator {
namespace example1 {

ConcreteColleagueA::ConcreteColleagueA(Mediator *mediator) : Colleague(mediator) {
}

void ConcreteColleagueA::someOperation() {
    //在需要跟其他同事通信的时候，通知中介者对象
    getMediator()->changed(this);
}
}
}
}
}
}
